openmole/plugins/org.openmole.plugin.tool.pattern/src/main/scala/org/openmole/plugin/tool/pattern/Strain.scala

```diff
@@ -28,7 +29,14 @@ object Strain {
     val first = Capsule(EmptyTask(), strainer = true)
     val last = Slot(Capsule(EmptyTask(), strainer = true))
 
-    (first -- puzzle -- last) + (first -- last)
+    val mole = puzzle.toMole
+    val sources = Sources(puzzle.sources.groupBy(_._1).mapValues(_.map(_._2)))
+    val hooks = Hooks(puzzle.hooks.groupBy(_._1).mapValues(_.map(_._2)))
+    val outputs = puzzle.lasts.foldLeft(PrototypeSet.empty) {
+      case (union, capsule) ⇒ union ++ capsule.outputs(mole, sources, hooks)
+    }.toSeq
```

Comment:
If you can modify/implement it with less verbose code that would be good of course. That's simply the best I came up with given the limited insights into the internals of OpenMOLE.